src/utils/transform.rs

```diff
@@ -0,0 +1,126 @@
+// this piece of code is needed only if "ignore_custom_section" is not set
+
+use crate::utils::module_info::ModuleInfo;
+use anyhow::{anyhow, Result};
+use wasm_encoder::SectionId;
+use wasmparser::{IndirectNameMap, NameMap, NameSectionReader};
+
+fn rebuild_name_map(name_map: NameMap) -> Result<wasm_encoder::NameMap> {
+	let mut encoded_map = wasm_encoder::NameMap::new();
+	for naming in name_map {
+		let naming = naming?;
+		encoded_map.append(naming.index, naming.name);
+	}
+	Ok(encoded_map)
+}
+
+fn rebuild_indirect_name_map(
+	indirect_name_map: IndirectNameMap,
+) -> Result<wasm_encoder::IndirectNameMap> {
+	let mut encoded_map = wasm_encoder::IndirectNameMap::new();
+	for indirect_naming in indirect_name_map {
+		let indirect_naming = indirect_naming?;
+		let new_map = rebuild_name_map(indirect_naming.names)?;
+
+		encoded_map.append(indirect_naming.index, &new_map);
+	}
+	Ok(encoded_map)
+}
+
+/// - Update function indices in Custom Name section Increment indices greater or equal than given
+///   one (incrementing by 1 because it is assumed one function has been added before the given
+///   index)
+/// - Rebuild remaining section items
+pub fn process_custom_section(
+	module_info: &mut ModuleInfo,
+	update_func_idx: Option<u32>,
```

Comment:
Can have some comments about the second argument?

My mental model is
```
enum FuncIdxTransformation {
    AsIs,
    IncrementByOneIfGreaterThan(u32),
    IncrementByOne
}
```

Also the behaviour if the index is already the MAX.

Reply:
Second argument `update_func_idx` is the starting index of the function indices that shall by incremented (because some function has been added before - in our case it is the gas function).

But I realized that:
- name `process_custom_section` is a bit misleading since we are just updating the function indices
-  if `update_func_idx` is None then no need to rebuild section at all